src/scriptProfiler.cpp

```diff
@@ -1215,9 +1240,72 @@ class GameInstructionNewExpression : public game_instruction {
 
 #pragma endregion Instructions
 #endif
+
+
+#if __linux__
+static std::filesystem::path sharedObjectPath;
+extern "C" int iterateCallback(struct dl_phdr_info* info, size_t size, void* data) {
+    std::filesystem::path sharedPath = info->dlpi_name;
+    if (sharedPath.filename() == "ArmaScriptProfiler_x64.so"sv) {
+        sharedObjectPath = sharedPath;
+        return 0;
+    }
+    return 0;
+}
+#endif
+
+std::filesystem::path getSharedObjectPath() {
+#if __linux__
+    dl_iterate_phdr(iterateCallback, nullptr);
+    return sharedObjectPath;
+
+#else
+    wchar_t buffer[MAX_PATH] = { 0 };
+    HMODULE handle = nullptr;
+    if(GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+            reinterpret_cast<LPCWSTR>(&getSharedObjectPath), &handle) == 0) {
+        sqf::diag_log("getSharedObjectPath: GetModuleHandle failed");
+        return std::filesystem::path{};
+    }
+
+    DWORD len = GetModuleFileNameW(handle, buffer, MAX_PATH);
+    if (len == 0) {
+        return std::filesystem::path{};
+    }
+    return std::filesystem::path{buffer};
+#endif
+}
+
+std::filesystem::path findConfigFilePath() {
+
+    std::filesystem::path path = getSharedObjectPath();
+    path = path.parent_path().parent_path();
+
+    std::filesystem::path dirName{ "config"sv };
+    std::filesystem::path fileName{ "parameters.json"sv };
+    path = path / dirName / fileName;
+
+    return path;
+}
+
 void scriptProfiler::preStart() {
     sqf::diag_log("Arma Script Profiler preStart");
 
+    std::filesystem::path filePath = findConfigFilePath();
+    
+    if (std::filesystem::exists(filePath)) {

```

Comment:
Check if no config path was found before call to fs? Otherwise some of the returns will try and check if current directory exists

Reply:
```suggestion
    if (!filePath.empty() && std::filesystem::exists(filePath)) {
```